Implement standalone web request tagging

stefaneberl opened the pull request:
This is handled similar to the way it's done in Java.
On a session it is possible to obtain a WebRequestTracer.

thomasgrassauer commented on src/openkit-shared/Core/NullRootAction.cs:
```diff
@@ -0,0 +1,12 @@
+﻿using Dynatrace.OpenKit.API;
```
license header

thomasgrassauer commented on src/openkit-shared/Core/NullRootAction.cs:
```diff
@@ -0,0 +1,12 @@
+
+namespace Dynatrace.OpenKit.Core
+{
+    public class NullRootAction : NullAction, IRootAction
```
document purpose of class

thomasgrassauer commented on src/openkit-shared/Core/Session.cs:
```diff
@@ -51,24 +53,44 @@ public Session(BeaconSender beaconSender, Beacon beacon)
         {
             // check if end() was already called before by looking at endTime
-            if (endTime != -1)
+            if (Interlocked.CompareExchange(ref endTime, beacon.CurrentTimestamp, -1L) != -1L)
```
In `WebRequestTracerBase.Stop` we check for `== 1L` and delegate to the `beaconSender `if the condition evaluates to `true`. Here we return on `!= -1L`. We should stick to only one pattern.

stefaneberl committed:
Implement standalone web request tagging

This is handled similar to the way it's done in Java.
On a session it is possible to obtain a WebRequestTracer.
